The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).PlanResourceChange call.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community ---> 


### Terraform Version

1.3.7

### AzureRM Provider Version

3.53.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app azurerm_linux_function_app azurerm_linux_function_app_slot

### Terraform Configuration Files

```hcl
Default configuration from https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_function_app
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.53.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 149 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice.(*LinuxFunctionAppDataSourceModel).unpackLinuxFunctionAppSettings(0xc001f03680, {{0xc0029d6000}, 0xc0029d4360, 0xc0029c2170, 0xc0029c2180, 0x0, 0xc0029c2190}, {0xc00198b000, {0x75cbf50, 0xc0004d5a70}, ...})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/linux_function_app_data_source.go:429 +0xc71
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice.LinuxFunctionAppDataSource.Read.func1({0x75cb7a8, 0xc002d37920}, {0xc00198b000, {0x75cbf50, 0xc0004d5a70}, 0xc000153e80, 0x0, {0x75cd0d0, 0xbaf04a8}})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/linux_function_app_data_source.go:366 +0x202a
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*DataSourceWrapper).DataSource.func2({0x75cb7a8, 0xc002d37920}, 0x1176592d574?, {0x6167c60?, 0xc00198b000?})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_data_source.go:48 +0x158
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x75cb7a8, 0xc002d37920}, 0x0?, {0x6167c60, 0xc00198b000})
	github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:185 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000235880, {0x75cb7e0, 0xc001e4eba0}, 0xd?, {0x6167c60, 0xc00198b000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000235880, {0x75cb7e0, 0xc001e4eba0}, 0xc000153d80, {0x6167c60, 0xc00198b000})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000240018, {0x75cb7e0?, 0xc001e4ea80?}, 0xc001cfee60)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1179 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0010c43c0, {0x75cb7e0?, 0xc0019cd0b0?}, 0xc000638d70)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:658 +0x403
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x68ed9a0?, 0xc0010c43c0}, {0x75cb7e0, 0xc0019cd0b0}, 0xc002449030, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004ea000, {0x75dadc0, 0xc00106e1a0}, 0xc001ea4ea0, 0xc0013d3140, 0xbaafaf0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0004ea000, {0x75dadc0, 0xc00106e1a0}, 0xc001ea4ea0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.53.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Plan: 16 to add, 0 to change, 0 to destroy.

### Actual Behaviour

 Error: Plugin did not respond
│ 
│   with module.service.module.web_app_slot.azurerm_linux_web_app_slot.slot,
│   on .terraform/modules/service/terraform/modules/azurerm/linux_web_app_slot/main.tf line 46, in resource "azurerm_linux_web_app_slot" "slot":
│   46: resource "azurerm_linux_web_app_slot" "slot" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵
╷
│ Error: Request cancelled
│ 
│   with module.service.data.azurerm_linux_function_app.function_apps["worker"],
│   on .terraform/modules/service/terraform/modules/services/web_api/function_app.tf line 48, in data "azurerm_linux_function_app" "function_apps":
│   48: data "azurerm_linux_function_app" "function_apps" {
│ 
│ The plugin.(*GRPCProvider).ReadDataSource request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.53.0_x5 plugin:


### Steps to Reproduce

1. terraform plan

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs

### Important Factoids

_No response_

### References

Similar issue in google plugin https://github.com/hashicorp/terraform-provider-google/issues/9030

--- a/internal/services/appservice/linux_function_app_data_source.go
+++ b/internal/services/appservice/linux_function_app_data_source.go
@@ -425,7 +425,7 @@ func (m *LinuxFunctionAppDataSourceModel) unpackLinuxFunctionAppSettings(input w
 			}
 		case "WEBSITE_HTTPLOGGING_RETENTION_DAYS":
 		case "FUNCTIONS_WORKER_RUNTIME":
-			if m.SiteConfig[0].ApplicationStack != nil {
+			if len(m.SiteConfig[0].ApplicationStack) > 0 {
 				m.SiteConfig[0].ApplicationStack[0].CustomHandler = strings.EqualFold(*v, "custom")
 			}
 

